### DISCUSSION TITLE
Roadmap to V2?

### DISCUSSION THREAD

**Discussion by michaelmerrill** [MichalLytek/type-graphql#1257] [General]
Hi @MichalLytek,

I've seen in a few of the issues and prs that you are working on a 2.0 for type-graphql. Is there a roadmap or issue we can follow to track the progress? Very excited!

**Comment by MichalLytek** [comment]
Due to all the circumstances, the 2.0 release will be just the v1.2.0-rc.1 with added GraphQL v16 support and code/deps cleanup.

The v3.0 release will be the big refactor upgrade with a lot of changes but no ETA yet.

**Comment by ericwooley** [comment]
Glad to hear that more major releases are coming!

Looks like 1.2.0-rc.1 was never actually released? Last release was 1.11 all the way back in November 2020.

Is 1.2.0-rc.1 stable enough to upgrade to for prod then? 

I know this is all open source work done for free, so I'm not trying to complain at all. But at the same time, I'd love to upgrade everything, that's been held back by the lack of gql 16 support. Maybe a an 2.0-rc would be possible?

**Reply by carlocorradini** [reply]
Already available:

![IMG_20230323_223016.jpg](https://user-images.githubusercontent.com/21266147/227368050-6dd5e53a-324c-4238-b9fb-762fb9a643d7.jpg)

Installation procedure:
```sh
npm uninstall type-graphql \
    && npm install type-graphql@next
```

**Reply by ericwooley** [reply]
huh, weird that it's not in the github release. 

Thanks!

**Reply by carlocorradini** [reply]
It is a release candidate. Not a proper "real" release. We are now improving toward a V2 release.
You can have more information following the issue https://github.com/MichalLytek/type-graphql/issues/1418

**Reply by MichalLytek** [reply]
> huh, weird that it's not in the github release.

1.2.0-rc.1 is in the github releases and npm:
https://github.com/MichalLytek/type-graphql/releases/tag/v1.2.0-rc.1
https://www.npmjs.com/package/type-graphql/v/1.2.0-rc.1

The `2.0.0-beta.1` is not on github releases as it was just a test publish to npm for the graphql `v16` upgrade.
When we merge #1400, we will be able to release more frequently the changes for the upcoming `v2.0` 💪
